example_external_provider_plugin/plugin.go

```diff
@@ -0,0 +1,215 @@
+package main
+
+import (
+	"context"
+	"time"
+
+	"google.golang.org/protobuf/types/known/timestamppb"
+
+	provider_service "github.com/openclarity/vmclarity/runtime_scan/pkg/provider/grpc/proto"
+
+	"github.com/openclarity/vmclarity/api/models"
+)
+
+type Provider struct {
+	provider_service.UnimplementedProviderServer
+}
+
+func (p *Provider) RunScanningJob(ctx context.Context, params *provider_service.RunScanningJobParams) (*provider_service.RunScanningJobResult, error) {
+	// Here need to run the VMClarity CLI and return the instance that is running the CLI.
+	return &provider_service.RunScanningJobResult{
+		Instance: &provider_service.Instance{
+			Id:           params.Id,
+			Location:     params.Location,
+			Image:        "ami-01d08089481510ba2",
+			InstanceType: "t2.large",
+			Platform:     "Linux/UNIX",
+			Tags: []*provider_service.Tag{
+				{
+					Key: "Name",
+					Val: "cloud provider plugin instance",
+				},
+			},
+			LaunchTime: timestamppb.New(time.Now()),
+		},
+	}, nil
+}
+
+func (p *Provider) DiscoverScopes(ctx context.Context, params *provider_service.DiscoverScopesParams) (*provider_service.DiscoverScopesResult, error) {
+	// Scopes can be very cloud provider specific. Here you see an example of an AWS scopes which
```

Comment:
I would replace this comments with:

TODO: Scopes are expected to be removed. Currently, the interface using the structure of the generated API scope types. Here we'll show an example of the existing AWS API models in order not to change the API.